### PR TITLE
DOC: API doc improvements

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is a follow-up to https://github.com/spacetelescope/jdaviz/pull/1333#discussion_r881682853 . Rather than just exposing the fitting stuff, might as well re-organize it so the most relevant API listed first, and also add more stuff to it.

Rendered doc at https://jdaviz--1346.org.readthedocs.build/en/1346/

p.s. I regret a little trying to clean things up... it quickly escalated.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-2459)

### CODE REVIEW

File: jdaviz/configs/default/plugins/model_fitting/fitting_backend.py [hunk 1]
```diff
@@ -15,31 +15,37 @@
...
     Parameters
     ----------
-    spectrum : :class:`specutils.spectra.Spectrum1D`
+    spectrum : `~specutils.Spectrum1D`
         The spectrum to be fitted.
+
     component_list : list
         Spectral model subcomponents stored in a list.
-        Their `name` attribute must be unique. Each subcomponent
-        should be an initialized object from `astropy.modeling.models'
```
Comment (pllim) [inline jdaviz/configs/default/plugins/model_fitting/fitting_backend.py:32]: The mismatch quotes on this line wasted an hour of my life. 🔥

**Comment by pllim** [comment]
I can reproduce the dev failure by rerunning a previously successful CI job in `main`. So it is caused by something upstream that has changed recently and unrelated to this PR.

**Review by kecnry — APPROVED** [review]
Woah, that did turn into a lot of changes!  The API organization I think will be quite useful, especially for new users to find their way around.  Thanks for cleaning this all up!

File: jdaviz/configs/default/plugins/model_fitting/initializers.py [hunk 2]
```diff
@@ -66,15 +66,15 @@ def initialize(self, instance, x, y):
 
         Parameters
         ----------
-        instance: `~astropy.modeling.models`
+        instance : `~astropy.modeling.Model`
```
[the remark this answers is not in the document]
Comment (pllim) [reply jdaviz/configs/default/plugins/model_fitting/initializers.py:69]: At least at one point in numpydoc world, you need spaces between the `:` or the API formatting is going to look wrong. Not all function docstrings are rendered, so I didn't go crazy with fixing that.

**Pull request merged**